eng/common/AzurePipelineTemplates/Maestro-PublishBuildToMaestro-Steps.yml

```diff
@@ -30,6 +30,39 @@ stages:
           Write-Host "##vso[task.complete result=Failed;]DONE"
       condition: or(eq(variables['MaestroUri'], ''), eq(variables['MaestroToken'], ''))
 
+# Parse Versions needed for offical CsWinRT version of build
+    - task: CmdLine@2
+      displayName: Parse Versions
+      inputs:
+        workingDirectory: $(Build.SourcesDirectory)
+        script: |
+          rem Parse the build-generated Build.BuildNumber into components that 
+          rem can be recombined for version resources, nuget packages, etc.
+   
+          @echo off
+   
+          rem Encode the build date/rev into a 16 bit value for resource versions
+          if "$(PrereleaseVersion)"=="" (
+            set RevisionBase=30000
+          ) else (
+            set RevisionBase=0
+          )
+          for /f "tokens=4,5 delims=." %%i in ("$(Build.BuildNumber)") do set BuildMetadata=%%i.%%j & set /a BuildRevision=%RevisionBase%+(((%%i/10000)-20)*366+((%%i)%%10000)/100*31+((%%i)%%100))*10+%%j

```

Comment:
I'm having a little trouble parsing what this expression is meant to produce. I'm not quite following the year & month calculations in build revision. My shell scripting is also a bit rusty, what does %%j represent?

Reply:
This is a bit arcane, this task gets us the version we're going to publish -- `1.4.2-prerelease.xxxxxx.x`
and this confusing bit calculates the `xxxxxx.x`, I believe. I think %%j might default to 0, but not sure. @Scottj1s  wrote this he might be able to elaborate.